skyportal/handlers/api/internal/image_analysis.py

```diff
@@ -0,0 +1,607 @@
+            )
+            pixscale = astrometry.get_pixscale(header=header)
+
+            log(
+                f'Frame center is {center_ra} {center_dec} radius {center_sr} deg, {pixscale*3600} arcsec/pixel'
+            )
+
+            # ## Reference catalogue
+            # Catalog name may be any Vizier identifier (ps1, gaiadr2, gaiaedr3, usnob1, gsc, skymapper, apass, sdss, atlas, vsx).
+            cat_refinement = catalogs.get_cat_vizier(
+                center_ra,
+                center_dec,
+                center_sr,
+                catalog_name_refinement,
+                filters={'rmag': '<21'},
+            )
+            log(f'{len(cat_refinement)} catalogue stars used for refinement')
+
+            cat_crossmatch = catalogs.get_cat_vizier(
+                center_ra,
+                center_dec,
+                center_sr,
+                catalog_name_crossmatch,
+                filters={'rmag': '<21'},
+            )
+            log(f'{len(cat_crossmatch)} catalogue stars used for crossmatch')
+
+            # ## Astrometric refinement
+            # Refining the astrometric solution based on the positions of detected objects and catalogue stars with scamp
+            wcs = pipeline.refine_astrometry(
+                obj,
+                cat_refinement,
+                wcs=wcs,
+                method=method,
+                cat_col_mag='rmag',
+                verbose=True,
+            )
+            if wcs is not None:
+                # Update WCS info in the header
+                astrometry.clear_wcs(
+                    header,
+                    remove_comments=True,
+                    remove_underscored=True,
+                    remove_history=True,
+                )
+                header.update(wcs.to_header(relax=True))
+
+            # ## Photometric calibration
+
+            # Positionally matching detected objects with catalogue stars
+            # Then building the photometric model for their instrumental magnitudes
+
+            # Photometric calibration using 2 arcsec matching radius, r magnitude, g-r color and second order spatial variations
+            m = pipeline.calibrate_photometry(
+                obj,
+                cat_refinement,
+                sr=matching_radius,
+                cat_col_mag='rmag',
+                cat_col_mag1='gmag',
+                cat_col_mag2='rmag',
+                max_intrinsic_rms=0.02,
+                order=2,
+                verbose=True,
+            )
+
+            # ## Simple catalogue-based transient detection
+            # Some transients may already be detected by comparing the detected objects with catalogue BUT limited approach
+
+            # Filtering of transient candidates
+            candidates = pipeline.filter_transient_candidates(
+                obj, cat=cat_crossmatch, sr=matching_radius, verbose=True
+            )
+            log(f'{candidates}')
```

Comment:
I'm definitely getting the feeling these log statements are for debugging and should not go to production as is.